src/node_modules/app/machines/explore/search.options.js

```diff
@@ -0,0 +1,289 @@
+import * as _ from 'lamb';
+import {get} from 'svelte/store';
+
+import {assign, send} from 'xstate';
+import {applyFnMap, makeIsIncluded} from '@svizzle/utils';
+import {getSearchURL} from 'utils/specs';
+import {request} from 'utils/net';
+import {
+	selectedDataset,
+	selectedDatasetSchema
+} from 'app/stores/exploreStores';
+import {stopWords} from 'utils/stopwords';
+import {createBaseSearchStores} from './search.context';
+
+// Fields utils
+const makeGetFieldsOfTypes = types => _.pipe([
+	_.pairs,
+	_.filterWith(_.pipe([
+		_.getPath('1.type'),
+		makeIsIncluded(types)
+	])),
+	_.mapWith(_.getAt(0))
+]);
+
+const getKeywordFields = makeGetFieldsOfTypes([
+	'keyword',
+	'keywordArray',
+	'textWithKeyword',
+	'textWithKeywordArray'
+]);
+
+function computeFieldSelection (fieldStats, userSelection) {
+	const exists = fieldStats.some(item => item.name === userSelection);
+	if (userSelection && exists) {
+		return userSelection;
+	}
+	if (fieldStats.length > 0) {
+		return fieldStats[0].name;
+	}
+	return null;
+}
+
+const computeCountQuery = (keywordFields, searchTerm) => ({
+	size: 0,
+	aggs : {
+		messages : {
+			filters : {
+				filters : _.fromPairs(keywordFields.map(name => [name, {
+					term : {
+						[name] : searchTerm
+					}
+				}]))
+			}
+		}
+	}
+});
+
+
+// Fields options
+const getCountInfo = applyFnMap({
+	name: _.getAt(0),
+	count: _.getPath('1.doc_count')
+});
+
+const mapResponseToFieldCount = _.pipe([
+	_.getPath('aggregations.messages.buckets'),
+	_.pairs,
+	_.mapWith(getCountInfo),
+	_.sortWith([_.sorterDesc(_.getKey('count'))]),
+]);
+
+const fieldsStatsRequest = ({searchQuery}) => {
+	const data = computeCountQuery(
+		getKeywordFields(get(selectedDatasetSchema)),
+		get(searchQuery)
+	);
+	const url = getSearchURL(get(selectedDataset));
+	return request('POST', url, {data});
+};
+
+const updateFieldStats = (ctx, {data}) => {
+	const fieldStats = mapResponseToFieldCount(data).filter(field =>
+		field.count > 0
+	);
```

Comment:
`filter` can be applied in `mapResponseToFieldCount` directly and before sorting:

```suggestion
const fieldsStatsRequest = ({searchQuery}) => {
	const data = computeCountQuery(
		getKeywordFields(get(selectedDatasetSchema)),
		get(searchQuery)
	);
	const url = getSearchURL(get(selectedDataset));
	return request('POST', url, {data});
};

const getCount = _.getKey('count');
const mapResponseToFieldCount = _.pipe([
	_.getPath('aggregations.messages.buckets'),
	_.pairs,
	_.mapWith(getCountInfo),
	_.filterWith(_.pipe([getCount, _.gt(0)])),
	_.sortWith([_.sorterDesc(getCount)]),
]);
const updateFieldStats = (ctx, {data}) => {
	const fieldStats = mapResponseToFieldCount(data);
```